src/components/BreedsPage/BreedsPage.tsx

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState, useCallback } from "react";
+import { Link, useNavigate, useParams } from "react-router-dom";
+
+import { getBreeds, IBreed } from "../../api";
+import { Spinner, CatDetailsModal } from "../shared";
+import { BreedDetailsModal } from "./components";
+
+import Styled from "./BreedsPage.styled";
+
+export const BreedsPage = () => {
+  const [breeds, setBreeds] = useState<IBreed[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const navigate = useNavigate();
+  const { breedId } = useParams<{ breedId: string; catId: string }>();
+
+  const fetchBreeds = useCallback(async () => {
+    setIsLoading(true);
+    try {
+      const response = await getBreeds();
+      setBreeds(response.data);
+    } catch (error) {
+      console.log(error);
+    } finally {
+      setIsLoading(false);
+    }
+  }, []);
+
+  const handleCloseModal = useCallback(() => {
+    navigate(`/breeds/${breedId}`);
+  }, [breedId, navigate]);
+
+  useEffect(() => {
+    fetchBreeds();
```

Comment:
Could you guess what would happen in these kind of cases if the component unmounts before the API response is returned?

Reply:
It will cause a memory leak.  We can add a check if the component is mounted or not inside our API call. A good practice here is to cancel requests on onMount or implement a custom hook to protect our API calls from memory leaks.

```
UseEffect(() => {
let isMounted = true;

const fetchBreeds = async() => {
...
getBreeds().then((response) => {
if(isMounted) {
  setBreeds(...)
}

fetchBreeds();

return () => {
  isMounted = false;

Reply:
👍  Correct - This will also throw a console error mentioning that you're trying to update state in an unmounted component.